dbt/include/athena/macros/materializations/incremental.sql

```diff
@@ -10,17 +22,52 @@
     );
 {%- endmacro %}
 
+{% macro delete_overlapping_partitions(target_relation, tmp_relation, partitioned_by) %}
+  {%- set partitioned_keys = partitioned_by | tojson | replace('\"', '') | replace('[', '') | replace(']', '') -%}
+  {% call statement('get_partitions', fetch_result=True) %}
+    select distinct {{partitioned_keys}} from {{ tmp_relation }};
+  {% endcall %}
+  {%- set table = load_result('get_partitions').table -%}
+  {%- set rows = table.rows -%}
+  {%- set partitions = [] -%}
+  {%- for row in rows -%}
+    {%- set single_partition = [] -%}
+    {%- for col in row -%}
+      {%- set column_type = adapter.convert_type(table, loop.index0) -%}
+      {%- if column_type == 'integer' -%}
+        {%- set value = col|string -%}
+      {%- elif column_type == 'string' -%}
+        {%- set value = "'" + col + "'" -%}
+      {%- elif column_type == 'date' -%}
+        {%- set value = "'" + col|string + "'" -%}
+      {%- else -%}
+        {%- do exceptions.raise_compiler_error('Need to add support for column type ' + column_type) -%}
+      {%- endif -%}
+      {%- do single_partition.append(partitioned_by[loop.index0] + '=' + value) -%}
+    {%- endfor -%}
+    {%- set single_partition_expression = single_partition | join(' and ') -%}
+    {%- do partitions.append('(' + single_partition_expression + ')') -%}
+  {%- endfor -%}
+  {%- set partition_expression = partitions | join(' or ') -%}
+    delete from
```

Comment:
Is `DELETE FROM` supported in Athena?
Running it manually on a table yields `[ErrorCategory:USER_ERROR, ErrorCode:NOT_SUPPORTED], Detail:NOT_SUPPORTED: Cannot delete from non-managed Hive table` in Athena (version 2).

Reply:
I believe these `delete from` statements are getting caught in the `execute` function and not actually being ran. 

I'm not a particular fan of this since it is it not clear. @tuan-seek is there a reason behind this pattern, why not just call `_clean_up_partitions` directly?

Reply:
The short answer is, I don't know if there is a way to call `_clean_up_partitions()` directly from macros. All Athena queries and AWS SDK calls are delegated back to `connections.py`. So I followed the same pattern here.

`DELETE FROM` is currently not supported in Athena. So the statement is intercepted in `connections.py` for us to do custom cleaning up and not issued towards Athena. If AWS is going to support `DELETE FROM` in the future, we can just clean up this custom implementation.

Reply:
@tuan-seek If you add functions to the Adapter class you can call them within Jinja. The Adapter class can be found within `imply.py` and can be called called as macro like `{% do adapter.your_function(your_parameters) %}` or `{{ adapter.drop_relation(existing_relation) }}`.

I will add a review pointing out what would be good to migrate to the Adapter class.

Reply:
Sounds good. Let me try that one!

Reply:
@tuan-seek Great and a big thanks for this PR, the functionality it could provide will be appreciated by so many, also apologies for taking so long to review!

Reply:
Likewise, thanks for starting this project. I've refactored the change as discussed. Please have a look when you have time @Tomme